File: aeongiftcardserver-product/src/main/java/com/luvina/pos/provider/domain/transaction/LogMaster.java
package com.luvina.pos.provider.domain.transaction;

import com.luvina.pos.provider.domain.base.AbstractAuditingEntity;
import com.luvina.pos.provider.domain.transaction.primarykey.LogMasterId;
import com.luvina.pos.provider.domain.transaction.primarykey.MachineStatusId;
import jakarta.persistence.Column;
import jakarta.persistence.EmbeddedId;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "log_master", schema = "T_KSM")
public class LogMaster extends AbstractAuditingEntity {
    @EmbeddedId
    private LogMasterId id;

    @Column(name = "ip_address", nullable = false)
    private String ipAddress;

    @Column(name = "mac_address", nullable = false)
    private String macAddress;

    @Column(name = "url", nullable = false)
    private String url;

    @Column(name = "headers", nullable = false)
    private String headers;

    @Column(name = "request", nullable = false)
    private String request;

    @Column(name = "response_code", nullable = false)
    private Integer responseCode;

    @Column(name = "response", nullable = false)
    private String response;

    @Column(name = "transaction_time", nullable = false)
    private String transactionTime;

    @Column(name = "environment", nullable = false)
    private String environment;

}

File: posserver-product-kasumi/src/main/kotlin/com/pos/system/model/db/LogMaster.kt
package com.pos.system.model.db

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.persistence.*
import java.sql.Timestamp

@Entity
@Table(name = "logmaster")
class LogMaster: BaseColumns() {
    @JsonProperty("company_code")
    @Column(table = "logmaster", name = "CompanyCode", length = 10)
    var companyCode: String? = null

    @JsonProperty("store_code")
    @Column(table = "logmaster", name = "StoreCode", length = 10)
    var storeCode: String? = null

    @JsonProperty("instore_code")
    @Column(table = "logmaster", name = "InstoreCode", length = 10)
    var instoreCode: String? = null

    @JsonProperty("ip_address")
    @Column(table = "logmaster", name = "IpAddress", length = 39)
    var ipAddress: String? = null

    @JsonProperty("url")
    @Column(table = "logmaster", name = "Url")
    var url: String? = null

    @JsonProperty("headers")
    @Column(table = "logmaster", name = "Headers")
    var headers: String? = null

    @JsonProperty("request")
    @Column(table = "logmaster", name = "Request")
    var request: String? = null

    @JsonProperty("response_code")
    @Column(table = "logmaster", name = "ResponseCode")
    var responseCode: Int? = null

    @JsonProperty("response")
    @Column(table = "logmaster", name = "Response")
    var response: String? = null

    @JsonProperty("mac_address")
    @Column(table = "logmaster", name = "MacAddress")
    var macAddress: String? = null

    @JsonProperty("record_timestamp")
    @Column(table = "logmaster", name = "RecordTimestamp")
    var recordTimestamp: Timestamp? = null

    @JsonProperty("environment")
    @Column(table = "logmaster", name = "Environment")
    var environment: String? = null

}